### PR TITLE
Usability fixes and improvements when using BUILD_STRATEGY_USE_HEX_FILE

### PR DESCRIPTION
**Pull request by oyvindronningstad**
- Fix an issue where the provision hex file was missing a public key digest when using MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE
- Add BUILD_STRATEGY options for the provision hex file
- Allow the BOOT_SIGNATURE_KEY_FILE path to be specified relative to the current sample directory

### CODE REVIEW

**Review by sigvartmh — APPROVED** [review]

**Comment by oyvindronningstad** [comment]
cc @hieuv

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -679,7 +679,15 @@ Bluetooth libraries and services
 Bootloader libraries
 --------------------
 
-|no_changes_yet_note|
+* :ref:`mcuboot_ncs`:
+
+  * Allow the path in :kconfig:option:`CONFIG_BOOT_SIGNATURE_KEY_FILE` to be relative to the current sample directory.
```
Comment (FrancescoSer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:684]:
```suggestion
  * The path set with :kconfig:option:`CONFIG_BOOT_SIGNATURE_KEY_FILE` can now be relative to the current sample directory.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -679,7 +679,15 @@ Bluetooth libraries and services
...
+    See the help text of the kconfig option for more details.
+
+* :ref:`bootloader`:
+
+  * Fixed: When both :ref:`bootloader` and :ref:`mcuboot_ncs` are present, and MCUboot is provided as a hex file (via :kconfig:option:`CONFIG_MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE`), the provision page was not regenerated correctly.
+    Fix generation of the provision page, and also add the option to provide the provision page as a hex file or, be skipped (:kconfig:option:`CONFIG_PROVISION_BUILD_STRATEGY_USE_HEX_FILE` and :kconfig:option:`CONFIG_PROVISION_BUILD_STRATEGY_SKIP_BUILD`)
```
Comment (FrancescoSer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:690]:
```suggestion
  * Added:

    * the :kconfig:option:`CONFIG_PROVISION_BUILD_STRATEGY_USE_HEX_FILE` Kconfig option to provide the provision page as a HEX file.
    * the :kconfig:option:`CONFIG_PROVISION_BUILD_STRATEGY_SKIP_BUILD` kconfig option to skip the building of the provision page. 
 
  * Fixed:

    * The generation of the provision page.
    * The regeneration of the provision page when both :ref:`bootloader` and :ref:`mcuboot_ncs` are present, and MCUboot is provided as a HEX file (using :kconfig:option:`CONFIG_MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE`).
```

This is a guess, let me know if the descriptions of the options mentioned are correct.

**Review by FrancescoSer — CHANGES_REQUESTED** [review]

File: modules/mcuboot/CMakeLists.txt [hunk 3]
```diff
@@ -308,10 +308,10 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
...
   if(IS_ABSOLUTE ${CONFIG_BOOT_SIGNATURE_KEY_FILE})
     set(mcuboot_key_file ${CONFIG_BOOT_SIGNATURE_KEY_FILE})
-  elseif (DEFINED mcuboot_CONF_DIR)
-    if (EXISTS ${mcuboot_CONF_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
-      set(mcuboot_key_file ${mcuboot_CONF_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
-    endif()
+  elseif (DEFINED mcuboot_CONF_DIR AND EXISTS ${mcuboot_CONF_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
+    set(mcuboot_key_file ${mcuboot_CONF_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
+  elseif (EXISTS ${CMAKE_SOURCE_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
+    set(mcuboot_key_file ${mcuboot_CONF_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
```
Comment (nordicjm) [inline modules/mcuboot/CMakeLists.txt:314]: This does not work and I do not like this change

File: subsys/bootloader/cmake/provision_hex.cmake [hunk 4]
```diff
@@ -27,9 +27,18 @@ if (CONFIG_NCS_IS_VARIANT_IMAGE)
...
+if (CONFIG_PROVISION_BUILD_STRATEGY_USE_HEX_FILE)
+  set(PROVISION_HEX ${CONFIG_PROVISION_HEX_FILE})
+  set(PROVISION_HEX_EXISTS TRUE)
+elseif(CONFIG_PROVISION_BUILD_STRATEGY_FROM_SOURCE)
+  # Build and include hex file containing provisioned data for the bootloader.
+  set(PROVISION_HEX_NAME     provision.hex)
+  set(PROVISION_HEX          ${PROJECT_BINARY_DIR}/${PROVISION_HEX_NAME})
+  set(GENERATE_PROVISION_HEX TRUE)
+  set(PROVISION_HEX_EXISTS   TRUE)
+endif()
```
Comment (nordicjm) [inline subsys/bootloader/cmake/provision_hex.cmake:41]: -1 on breaking sysbuild support, we need to stop making changing to child/parent images

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by FrancescoSer — APPROVED** [review]

**Comment by CodingGhost** [comment]
can this be reopened? being able to specify the hex file and signature file relative to the working directory is very Important for us. Currently we have to use very awkward workarounds for that.

**Comment by nordicjm** [comment]
> @nordicjm @oyvindronningstad can this be reopened? being able to specify the hex file and signature file relative to the working directory is very Important for us. Currently we have to use very awkward workarounds for that.

This will not work with sysbuild and it's possible that the whole feature will be dropped in it's entirety

**Comment by CodingGhost** [comment]
@nordicjm and is there any other possibility to specify the file relatively?
I think my use case for this is pretty common. multiple developers trying to build the project on their own PCs, for every build one has to change the path of the hex file manually... This is bad.

**Comment by CodingGhost** [comment]
any update on this? How do people solve this issue or work around it?

**Pull request closed without merging**